web_refresher/static/src/js/refresher.esm.js

```diff
@@ -1,16 +1,64 @@
 /** @odoo-module **/
 /* Copyright 2022 Tecnativa - Alexandre D. Díaz
  * Copyright 2022 Tecnativa - Carlos Roca
+ * Copyright 2023 Taras Shabaranskyi
  * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
 
-const {Component} = owl;
+import {Component} from "@odoo/owl";
+import {_t} from "@web/core/l10n/translation";
+import {useService} from "@web/core/utils/hooks";
 
 export class Refresher extends Component {
-    _doRefresh() {
-        // Note: here we use the pager props, see xml
-        const {limit, offset} = this.props;
-        this.props.onUpdate({offset, limit});
+    setup() {
+        super.setup();
+        this.notification = useService("notification");
+    }
+
+    /**
+     * @returns {Boolean}
+     */
+    get displayButton() {
+        const {searchModel, pagerProps} = this.props;
+        const hasSearchModel = searchModel && searchModel.search;
+        return Boolean(hasSearchModel || (pagerProps && pagerProps.onUpdate));
+    }
+
+    _searchModelRefresh() {
+        const {searchModel} = this.props;
+        if (searchModel && typeof searchModel.search === "function") {
+            searchModel.search();
+        }
+    }
+
+    async _pagerRefresh() {
+        const pagerProps = this.props.pagerProps;
+        if (pagerProps && typeof pagerProps.onUpdate === "function") {
+            const {limit, offset} = pagerProps;
+            await pagerProps.onUpdate({offset, limit});
+        }
+    }
+
+    async refresh() {
+        const {searchModel, pagerProps} = this.props;
+        if (searchModel) {
+            this._searchModelRefresh();
+        } else if (pagerProps) {
+            await this._pagerRefresh();
+        }
+    }
+
+    async onClickRefresh() {
+        await this.refresh();
+        this.notification.add(_t("Refreshed"), {
```

Comment:
No very convinced with this :) Is it really needed?

Reply:
> No very convinced with this :) Is it really needed?

Are you talking about t**his.notification.add** or the entire component?

Reply:
Yes, just the notification of the refresh :)

Reply:
this is not necessary, but I noticed in my experience that if you do not display anything, it is not immediately clear whether the click has worked, in a situation where there were no new changes to update

Reply:
If you think it's unnecessary or distracting, I can remove it

Reply:
Maybe a shorter notification close timeout would be less anoying

Reply:
ok, I'll think about a simplified version of the notification, as a last resort I'll just remove it

Reply:
![ezgif com-video-to-gif](https://github.com/OCA/web/assets/17133203/5bb21cbf-0ab9-43b6-b206-81a15eb6c0e9)
What about this animation instead of notification (the red ring is added by the screen recorder)

> Maybe a shorter notification close timeout would be less anoying